src/commands/soundbot/MostPlayedCommand.ts

```diff
@@ -0,0 +1,45 @@
+import { Message } from 'discord.js';
+
+import BaseCommand from '../base/BaseCommand';
+
+import DatabaseAdapter from '../../db/DatabaseAdapter';
+
+export class MostPlayedCommand extends BaseCommand {
+  private db: DatabaseAdapter;
+
+  constructor(message: Message, db: DatabaseAdapter) {
+    super(message);
+    this.db = db;
+  }
+
+  public run() {
+    this.message.channel.send(this.getMostPlayedSounds());
+  }
+
+  private getMostPlayedSounds() {
+    const sounds = this.db.getMostPlayedSounds();
+    return this.prepareMessageFromSounds(sounds);
+  }
+
+  private prepareMessageFromSounds(sounds: Array<{ name: string, count: number }>) {
```

Comment:
Looks like you never specify the return type. Is it not required by typescript?

Reply:
The returned type is perfectly inferred by TypeScript. Following DRY here haha